src/instructlab/sdg/llmblock.py

```diff
@@ -84,27 +90,40 @@ def _parse(self, generated_string) -> dict:
 
         return matches
 
+    def _format_prompt(self, sample: Dict) -> str:
+        return self.prompt_template.format(**sample).strip()
+
     def _generate(self, samples, **gen_kwargs) -> list:
         prompts = [
-            self.model_prompt.format(
-                prompt=self.prompt_template.format(**sample).strip()
-            )
+            self.model_prompt.format(prompt=self._format_prompt(sample))
             for sample in samples
         ]
-        response = self.client.completions.create(
-            prompt=prompts, **{**self.defaults, **gen_kwargs}
-        )
-        return [choice.text.strip() for choice in response.choices]
+        generate_args = {**self.defaults, **gen_kwargs}
+
+        if self.server_supports_batched:
+            response = self.client.completions.create(
+                prompt=prompts, **generate_args
+            )
+            return [choice.text.strip() for choice in response.choices]
+
+        n = gen_kwargs.get("n", 1)
```

Comment:
I would have imagined doing this in reverse - including "num_instructions_to_generate" in the block config and adding 'n' to gen_kwargs if batching was supported. No biggie though